types/ramda/index.d.ts

```diff
@@ -888,6 +888,7 @@ export function identity<T>(a: T): T;
  */
 // tslint:disable:max-line-length
 export function ifElse<TArgs extends any[], TOnTrueResult, TOnFalseResult>(fn: (...args: TArgs) => boolean, onTrue: (...args: TArgs) => TOnTrueResult, onFalse: (...args: TArgs) => TOnFalseResult): (...args: TArgs) => TOnTrueResult | TOnFalseResult;
+export function ifElse<TArgs extends any, N extends TArgs, TOnTrueResult, TOnFalseResult>(fn: (args: TArgs) => args is N, onTrue: (args: N) => TOnTrueResult, onFalse: (args: Exclude<TArgs, N>) => TOnFalseResult): (args: TArgs) => TOnTrueResult | TOnFalseResult;
```

Comment:
`ifElse` support multiple arguments, but this function type only has one argument.

Why `onFalse` must exclude N ? The type should be common enough to cover most case.